modules/issue_tracker/php/NDB_Menu_Filter_Form_issue_tracker.class.inc

```diff
@@ -0,0 +1,384 @@
+            array(
+                "size" => 10,
+                "maxlength" => 50,
+            )
+        );
+        $this->addBasicText(
+            'issueID',
+            'Issue ID',
+            array(
+                "size" => 10,
+                "maxlength" => 25,
+            )
+        );
+        $this->addSelect('module', 'Module', $modules);
+        $this->addSelect('category', 'Category', $categories);
+        $this->addBasicText(
+            'PSCID',
+            'PSCID',
+            array(
+                "size" => 10,
+                "maxlength" => 25,
+            )
+        );
+        $this->addBasicText(
+            'visitLabel',
+            'Visit Label',
+            array(
+                'size' => "5",
+                "maxlength" => "15",
+            )
+        );
+        $this->addSelect('site', 'Site', $list_of_sites);
+        $this->addSelect('reporter', 'Reporter', $reporters);
+        $this->addSelect('assignee', 'Assignee', $assignees);
+        $this->addSelect('priority', 'Priority', $priorities);
+        $this->addSelect('status', 'Status', $statuses);
+        $this->addBasicDate('minDate', 'Created After', $dateOptions);
+        $this->addBasicDate('maxDate', 'Created Before', $dateOptions);
+        $this->addCheckbox('watching', 'Watching', array('value' => '1'));
+
+        $this->tpl_data['backURL'] = $_SERVER['REQUEST_URI'];
+    }
+
+
+    /**
+     * Overwrites the function to add a customized filter
+     * for watching and include closed
+     *
+     * @param string $prepared_key filter key
+     * @param string $field filter field
+     * @param string $val filter value
+     *
+     * @return null
+     */
+    function _addValidFilters($prepared_key, $field, $val)
+    {
+        $user =& User::singleton();
+        $query = '';
+        if ((!empty($val) || $val === '0') && $field != 'order') {
+            if (($field != 'w.userID')
+                && ($field != 'minDate')
+                && ($field != 'maxDate')
+            ) {
+                if (in_array($field, $this->CheckboxFilters) && $val) {
+                    $query .= " AND $field";
+                } elseif (isset($this->EqualityFilters)
+                    && in_array($field, $this->EqualityFilters)
+                ) {
+                    $query .= " AND $field = :v_$prepared_key";
+                } else {
+                    $query .= " AND $field LIKE CONCAT('%', :v_$prepared_key, '%')";
+                }
+            } else if ($field = 'w.userID' && ($val == 1)) {
+                $query .= " AND w.userID = :v_$prepared_key";
+            } else if ($field = 'w.userID' && ($val == 0)) {
+                    $query .= " AND w.userID <> :v_$prepared_key";
+            } else if ($field = 'minDate') {
+                $query .= " AND i.dateCreated > '" . $val . "00-00-00'";
+            } else if ($field = 'maxDate') {
+                $query .= " AND i.dateCreated < " . $val . "00-00-00'";
```

Comment:
* replace $val with a prepared key - user-input value shouldn't be passed directly
* maxDate filter broken - probably due to missing single quote after i.dateCreated <